scripts/create_dataset.py

```diff
@@ -1,69 +1,161 @@
+    for input_column in input_table["columns"]:
+        column, relationships = create_column_and_relationships(input_relationships=input_column.get("relationships"),
+                                                                column_name=input_column["name"],
+                                                                column_datatype=input_column["datatype"],
+                                                                column_array_of=input_column.get("array_of"),
+                                                                input_table_name=input_table_name,
+                                                                input_tables=input_tables)
+        columns.append(column)
+        table_relationships.extend(relationships)
+
+    table = {"name": input_table_name, "columns": columns}
+
+    return table, table_relationships
+
+
+def create_column_and_relationships(input_relationships,
+                                    column_name,
+                                    column_datatype,
+                                    column_array_of,
+                                    input_table_name,
+                                    input_tables):
+    """
+    Creates a column and relationships, keeping ony the name, data type and column field from the input column.
+    :param input_relationships: the list of relationships from an input JSON column
+    :param column_name: the name of the input JSON column
+    :param column_datatype: the primitive type enforced for values of the input JSON column
+    :param column_array_of: indicates if the column values should be in array
+    :param input_table_name: the name of input JSON table that has the input_column
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    relationships = [
+        {
+            "name": "{}_{}_to_{}_{}".format(input_table_name,
+                                            column_name,
+                                            input_relationship["table_name"],
+                                            input_relationship["column_name"]),
+            "from": {"table": input_table_name, "column": column_name, "cardinality": "one"},
+            "to": {
+                    "table": input_relationship["table_name"],
+                    "column": input_relationship["column_name"],
+                    "cardinality": "many" if column_array_of is True else "one"
+                  }
+        }
+        if validate_table_and_column(
+               table_name=input_table_name,
+               column_name=column_name,
+               input_tables=input_tables) and validate_table_and_column(table_name=input_relationship["table_name"],
+                                                                        column_name=input_relationship["column_name"],
+                                                                        input_tables=input_tables)
```

Comment:
I'd break these two `validate_table_and_column` calls into local variables before the `if`, to make this condition less gnarly

Reply:
Oh wait, this is inside the list comprehension...

I think this is complex enough to justify doing a few loops over the input relationships, instead of trying to do it all in one shot.